src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -294,24 +298,27 @@ internal static ProcessInfo[] GetProcessInfos(Predicate<int>? processIdFilter =
                                     throw new InvalidOperationException(SR.CouldntGetProcessInfos, new Win32Exception((int)status));
                                 }
 
+                                MostRecentSize = buffer.Length;
```

Comment:
`buffer.Length` can be a lot more than what's required.

The estimate for the next buffer size should be `requiredSize` plus some slack space to allow for fluctuations in the process list, similar to what `GetNewBufferSize` does.

Reply:
This still needs to be addressed.

Reply:
> This still needs to be addressed.

@jkotas Done, PTAL